src/goto-instrument/code_contracts.cpp

```diff
@@ -282,24 +335,388 @@ const symbolt &code_contractst::new_tmp_symbol(
     symbol_table);
 }
 
+exprt create_alias_expression(
+  const exprt &assigns,
+  const exprt &lhs,
+  std::vector<exprt> &aliasable_references)
+{
+  bool first_iter = true;
+  exprt running = false_exprt();
```

Comment:
This can be simplified by using and object of type `exprt::operandst` with each call to `same_object` doing `operands.push_back(same_object(leftPtr, rightPtr);` and then doing `return disjunction(operands);` at the very end.